newIDE/app/src/VariablesList/VariablesList.js

```diff
@@ -201,7 +201,7 @@ const VariableRow = React.memo<VariableRowProps>(
     isSelected,
     onSelect,
     gdevelopTheme,
-    topLevelVariableNameInputRefs,
+    variableNameInputRefs,
     topLevelVariableValueInputRefs,
```

Comment:
it feel weird having all variable names but only top level values?

Reply:
It's not weird if you consider what they're used for:
- All the names must be stored because the React key for the row contains the variable name (this could be changed) so if you change the name, you need the reference to the input in order to refocus the field
- The same challenge happens with values, but only for top level ones in the case you want to modify the value at the level instance of an object's variable: in that case we create a new variable and need to focus the new variable value field.
Do you feel like I should mention those reasons in the code as a comment?

Reply:
I think it wouldn't hurt, to avoid this discussion next time!